src/components/Prompt.tsx

```diff
@@ -57,16 +69,29 @@ export function Prompt({
       .getElementById("promptButtons")
       ?.scrollIntoView(/* { behavior: "smooth" } */);
 
-    const promptBox = window.document.getElementById(
-      "promptBox"
-    ) as HTMLTextAreaElement | null;
+    // If the promptNode swap is not being triggered by a click.
+    if (textOffsetRef.current === -1) {
+      // Reset editing state.
+      setIsEditing(
+        promptNodeType === FluxNodeType.User || promptNodeType === FluxNodeType.System
+      );
+    }
 
-    // Focus the text box and move the cursor to chosen offset (defaults to end).
-    promptBox?.setSelectionRange(textOffsetRef.current, textOffsetRef.current);
-    promptBox?.focus();
+    // TODO: Really wish we didn't have to do this...
+    // is there an optimization we can make somewhere?
+    // Need a small timeout to ensure the text box is rendered.
+    setTimeout(() => {
+      const promptBox = window.document.getElementById(
+        "promptBox"
+      ) as HTMLTextAreaElement | null;
 
-    // Default to moving to the start of the text.
-    textOffsetRef.current = -1;
+      // Focus the text box and move the cursor to chosen offset (defaults to end).
+      promptBox?.setSelectionRange(textOffsetRef.current, textOffsetRef.current);
+      promptBox?.focus();
+
+      // Default to moving to the start of the text.
+      textOffsetRef.current = -1;
+    }, 50);
```

Comment:
I guess that you need to compose `TextareaAutosize` in another component, let's say `PromptBox`, and use `useEffect` inside it, so you ensure the effect triggers specifically on this text area mount.

It's unfortunate that `react-textarea-autosize` library doesn't handle refs...

Reply:
ah i figured out the issue, its not that there's any lag in rendering its cuz isEditing is set in the useEffect is what actually renders the textbox... fixing